src/logstatsexp.jl

```diff
@@ -0,0 +1,38 @@
+"""
+    logmeanexp(A; dims=:)
+
+Computes `log.(mean(exp.(A); dims))`, in a numerically stable way.
+"""
+function logmeanexp(A::AbstractArray; dims=:)
+    R = logsumexp(A; dims)
+    N = length(A) ÷ length(R)
+    return R .- log(N)
+end
+
+"""
+    logvarexp(A; dims=:)
+
+Computes `log.(var(exp.(A); dims))`, in a numerically stable way.
+"""
+function logvarexp(
+    A::AbstractArray; dims=:, corrected::Bool=true, logmean=logmeanexp(A; dims)
+)
+    R = logsumexp(2logsubexp.(A, logmean); dims)
+    N = length(A) ÷ length(R)
+	if corrected
+		return R .- log(N - 1)
```

Comment:
Again this causes undesired promotions and allocations.

Reply:
The unnecessary allocations are still present.

Reply:
```julia
R .- log(N - 1)
```
creates a new array. But if `R` is e.g. of type `Array` this causes unnecessary allocations.